src/GLFW.jl

```diff
@@ -22,8 +22,23 @@ else
 	error("GLFW $VERSION is not supported")
 end
 
+immutable GLFWError <: Exception
+    code::ErrorCode
+    description::UTF8String
+end
+function Base.showerror(io::IO, e::GLFWError)
+    print(io,
+        "GLWF reported an error with the code: ",
```

Comment:
I would have to play with it a bit, but I suspect the enum name (which is what gets interpolated into a string) and the description are going to be redundant, so you'll wind up with messages like, `GLFWError with code: NOT_INITIALIZED: GLFW has not been initialized`.

👍 as long as it's on one line.